sys/net/application_layer/dhcpv6/client.c

```diff
@@ -524,6 +648,41 @@ static void _schedule_t1_t2(void)
     _schedule_t2();
 }
 
+static void _update_t1_t2(unsigned lease_t1, unsigned lease_t2)
+{
+    if ((lease_t1 != 0) && (lease_t2 != 0) &&
+        ((server.t1 == 0) || (server.t1 >= lease_t1)) &&
+        ((t2 == 0) || (t2 >= lease_t2))) {
+        server.t1 = lease_t1;
+        t2 = lease_t2;
+        _schedule_t1_t2();
+    }
+}
+
+static void _update_t2(unsigned lease_t1, unsigned lease_t2)
+{
+    if ((lease_t1 != 0) && (lease_t2 != 0) &&
+        (server.t1 > lease_t1) && (t2 > lease_t2)) {
+        server.t1 = lease_t1;
+        t2 = lease_t2;
+        _schedule_t2();
+    }
+}
+
+static void _update_addr_lease(const dhcpv6_opt_iaaddr_t *iaaddr, addr_lease_t *lease) {
+    if (iaaddr != NULL) {
+        uint32_t valid = byteorder_ntohl(iaaddr->valid);
+        uint32_t pref = byteorder_ntohl(iaaddr->pref);
+
+        lease->leased = 1U;
+        memcpy(&lease->addr, &iaaddr->addr, sizeof(ipv6_addr_t));
+        dhcpv6_client_conf_prefix(
+                lease->parent.ia_id.info.netif, &lease->addr,
+                128, valid, pref
```

Comment:
Try to prevent magic numbers. We have `IPV6_ADDR_BIT_LEN` for this.
```suggestion
                IPV6_ADDR_BIT_LEN, valid, pref
```

Reply:
Oh, very good, thank you!